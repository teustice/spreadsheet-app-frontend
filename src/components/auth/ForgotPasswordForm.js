import React, {Component} from 'react';
import { connect } from 'react-redux';

import {
  getUsers
} from '../../actions/userActions'
import apiUrl from '../../lib/apiUrl';

class ForgotPasswordForm extends Component {
    constructor(props) {
        super(props);

        this.state = {
            email: '',
            showForm: false,
            errors: undefined,
        }
    }

    forgotPassword(e) {
      e.preventDefault();
      let body;
      let that = this;

      if(this.state.email){
        body = {
          email: this.state.email
        }
        fetch(`${apiUrl}/users/forgot`, {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(body)
        })
            .then(function(res) {
                if(res.ok) {
                  that.setState({errors: {error: `Email to reset password has been sent to ${this.state.email}`}})
                  that.setState({email: ''})
                } else {
                  that.setState({errors: res})
                }
            })
            .catch(error => console.error('Error:', error));

        if(this.props.submitCallback) {
          this.props.submitCallback();
        }
      }
    }

    updateEmail(e) {
        this.setState({email: e.target.value})
    }


    render() {
      let errors = [];
      if(this.state.errors) {
          for(var error in this.state.errors){
            console.log(this.state.errors[error]);
              errors.push(
                  <p key={this.state.errors[error]}>{this.state.errors[error]}</p>
              )
          }
      }
        return (
            <div className="forgot-password-form">

              <form onSubmit={this.forgotPassword.bind(this)} >
                <div className="form-group">
                  <div className="input-wrapper">
                    <label>Email</label>
                    <input name="email" type='text' value={this.state.email} onChange={this.updateEmail.bind(this)} required />
                  </div>
                  <button type='submit' className="btn btn-sm">Reset Password</button>
                </div>
                <div>{errors}</div>
              </form>
            </div>
        );
    }
}


const mapStateToProps = state => ({
 ...state
})

const mapDispatchToProps = dispatch => ({
  getUsers: (token) => dispatch(getUsers(token))
})


export default connect(mapStateToProps, mapDispatchToProps)(ForgotPasswordForm);
