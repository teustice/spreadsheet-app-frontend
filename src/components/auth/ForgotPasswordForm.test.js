import React from "react";
import {configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import {ForgotPasswordForm} from "./ForgotPasswordForm";

configure({ adapter: new Adapter() });

describe('<ForgotPasswordForm />', () => {
  describe('render()', () => {

    test('renders the component', () => {
      expect(toJson(<ForgotPasswordForm />)).toMatchSnapshot();
    });

  });
});
