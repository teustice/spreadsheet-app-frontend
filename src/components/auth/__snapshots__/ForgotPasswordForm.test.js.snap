// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPasswordForm /> render() renders the component 1`] = `undefined`;
